File: LocalPayment/src/main/java/com/braintreepayments/api/LocalPaymentInternalAuthRequestCallback.java
package com.braintreepayments.api;

import androidx.annotation.Nullable;

/**
 * Callback for receiving result of
 * {@link LocalPaymentClient#createPaymentAuthRequest(LocalPaymentRequest, LocalPaymentInternalAuthRequestCallback)}.
 */
public interface LocalPaymentInternalAuthRequestCallback {

    /**
     * @param localPaymentAuthRequestParams {@link LocalPaymentAuthRequestParams}
     * @param error an exception that occurred while initiating a Local Payment
     */
    void onResult(@Nullable LocalPaymentAuthRequestParams localPaymentAuthRequestParams, @Nullable Exception error);
}

File: LocalPayment/src/main/java/com/braintreepayments/api/localpayment/LocalPaymentInternalAuthRequestCallback.kt
package com.braintreepayments.api.localpayment

/**
 * Callback for receiving result of [LocalPaymentClient.createPaymentAuthRequest].
 */
internal fun interface LocalPaymentInternalAuthRequestCallback {

    /**
     * @param localPaymentAuthRequestParams [LocalPaymentAuthRequestParams]
     * @param error an exception that occurred while initiating a Local Payment
     */
    fun onLocalPaymentInternalAuthResult(
        localPaymentAuthRequestParams: LocalPaymentAuthRequestParams?,
        error: Exception?
    )
}
